tests/core/tax_scales/test_abstract_rate_tax_scale.py

```diff
@@ -0,0 +1,71 @@
+from numpy import array
+
+from openfisca_core.taxscales import AbstractRateTaxScale
+from openfisca_core.tools import assert_near
+
+
+def test_bracket_indices():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base)
+
+    assert_near(result, [0, 0, 0, 1, 1, 2])
+
+
+def test_bracket_indices_with_factor():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base, factor = 2.0)
+
+    assert_near(result, [0, 0, 0, 0, 1, 1])
+
+
+def test_bracket_indices_with_round_decimals():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base, round_decimals = 0)
+
+    assert_near(result, [0, 0, 0, 1, 1, 2])
```

Comment:
@benjello I expected this test to pass, What's the expected result?

Reply:
I fixed that test. Barckect intervals are right-open.